### PR TITLE
Enable functions, procedures, and triggers

### PR DESCRIPTION
**Pull request by timothy-spencer**
This is to satisfy https://github.com/18F/cg-product/issues/892

The documentation update should show you how to enable the feature flags, and then how to use them.

I have tested this out by building and deploying this in my GSA cloud.gov sandbox account, pointing at the 18F sandbox AWS account.  These are the things that I did to test:
- [x] 	cf create-service aws-rds medium-mysql test3 -b tspencer-awsbroker -c '{"enable_functions": true, "publicly_accessible": true}'
- [x] 	cf bind-service testapp test3
- [x] 	cf connect-to-service testapp test3
- [x] 	create function hello(id INT) returns CHAR(50) return 'foobar'; # should give Query OK
- [x] 	cf unbind-service testapp test3
- [x] 	cf delete-service test3 # should delete db
- [x] 	wait until test3 is gone
- [x] 	cf delete-service <anotherdb> # should delete test3’s pgroup
- [x] 	cf create-service aws-rds medium-mysql test3 -b tspencer-awsbroker -c '{"enable_functions": false, "publicly_accessible": true}'
- [x] 	cf bind-service testapp test3
- [x] 	cf connect-to-service testapp test3
- [x] 	create function hello(id INT) returns CHAR(50) return 'foobar';  # <should give error, instead of Query OK>
- [x] 	turn off public access feature flag and restage
- [x] 	cf create-service aws-rds medium-mysql test3 -b tspencer-awsbroker -c '{"enable_functions": true, "publicly_accessible": true}'
- [x] 	should result in db that is not public
- [x] 	turn off enable functions feature flag and restage
- [x] 	turn on public access feature flag and restage
- [x] 	cf create-service aws-rds medium-mysql test3 -b tspencer-awsbroker -c '{"publicly_accessible": true}'
- [x] 	cf bind-service testapp test3
- [x] 	cf connect-to-service testapp test3
- [x] 	create function hello(id INT) returns CHAR(50) return 'foobar';  # <should give error, instead of Query OK>

### CODE REVIEW

**Comment by timothy-spencer** [comment]
One last question: Would you like me to create another PR to enable the feature, or leave that to you? I kind of assumed you would want to do some independent testing, so didn't enable any of the functionality by default, but I realized that if you wanted this to be a largely zero-effort thing on your part, you might want the concourse/deployment aspect addressed too.

**Comment by timothy-spencer** [comment]
Happy to pair on this if you want a tour, but it does take a while to provision databases, so if you give me fair warning, I can have a pair provisioned with and without the config option to demo.

**Comment by mxplusb** [comment]
@timothy-spencer I will take a look at this today and get back to you. Would be good to pair at some point.

File: services/rds/rds.go [hunk 1]
```diff
@@ -219,6 +314,40 @@ func (d *dedicatedDBAdapter) bindDBToApp(i *RDSInstance, password string) (map[s
...
 
+// search out all the parameter groups that we created and try to clean them up
+func cleanupCustomParameterGroups(svc *rds.RDS) {
+	input := &rds.DescribeDBParameterGroupsInput{}
+	err := svc.DescribeDBParameterGroupsPages(input,
+		func(pgroups *rds.DescribeDBParameterGroupsOutput, lastPage bool) bool {
+			// If the pgroup matches the prefix, then try to delete it.
+			// If it's in use, it will fail, so ignore that.
+			for _, pgroup := range pgroups.DBParameterGroups {
+				if matched, _ := regexp.Match("^"+PgroupPrefix, []byte(*pgroup.DBParameterGroupName)); matched {
```
Comment (mxplusb) [inline services/rds/rds.go]: Can you add error handling here? I don't want to swallow errors.

Comment (timothy-spencer) [reply services/rds/rds.go]: I guess we could log an error.  Good call!  Will have that in shortly.

Comment (timothy-spencer) [reply services/rds/rds.go]: done!

**Comment by timothy-spencer** [comment]
BTW, I have some changes ready to commit that should enable this feature and test it out.  Would you like that to be part of this PR, or a separate one, or for you to do that yourself?

**Comment by mxplusb** [comment]
Go ahead and add them, thanks!

**Comment by timothy-spencer** [comment]
Done.  I can't test those out, because I don't have concourse and all that jazz going, but it should work.

**Comment by mogul** [comment]
At a quick glance, yes... I'll merge. Unclear to me: will the next deploy have the new features enabled?

**Comment by timothy-spencer** [comment]
The next deploy should have the functions enabled:  https://github.com/18F/aws-broker/pull/48/commits/10dbb17e844449d9e7ac68fc5a993f633812f200

**Comment by mxplusb** [comment]
Yeah. Terraform is complaining, I will have to fix it before this finalizes. Gimme a couple hours to work on it.

**Comment by mxplusb** [comment]
@timothy-spencer deploying the boker with your changes threw this error:

```
2019-06-12T13:30:05.43-0600 [APP/PROC/WEB/0] ERR ERROR 1418 (HY000) at line 1: This function has none of DETERMINISTIC, NO SQL, or READS SQL DATA in its declaration and binary logging is enabled (you *might* want to use the less safe log_bin_trust_function_creators variable)
2019-06-12T13:30:05.45-0600 [APP/PROC/WEB/0] OUT Exit status 1
```

**Comment by timothy-spencer** [comment]
Hmm.  Is that the test?  Dang it!  I just realized that we also need to have the -c '{"enable_functions": true}' option set when creating the db with that option.  This is something that I couldn't test without concourse and so on.  

That is actually proper, because the broker is creating a db, but the functions are not enabled.  I will see if I can figure out where to do that.

**Pull request merged**